app/views/admin/products/_form.html.haml

```diff
@@ -99,4 +99,6 @@
 
     .form_group#submit_div
       = f.submit class: 'btn btn-success'
-      = button_tag 'Cancel', type: :button, onclick: 'confirmProductCancel()', id: :cancel_button, class: 'btn btn-danger'
\ No newline at end of file
+      -if @product.persisted?
```

Comment:
What is the purpose of this call to persisted? I'm sure its necessary, but I'm unclear why.

Reply:
This checks to see if the product/vendor is already created or not. If it is, then we want to show the delete button, if not, then the cancel button suffices on a create new form.